src/BenchmarkDotNet/Engines/Engine.cs

```diff
@@ -43,13 +44,14 @@ public class Engine : IEngine
         private readonly EngineWarmupStage warmupStage;
         private readonly EngineActualStage actualStage;
         private readonly bool includeMemoryStats;
+        private readonly ICacheClearingStrategy cacheClearingStrategy;
 
         internal Engine(
             IHost host,
             IResolver resolver,
             Action dummy1Action, Action dummy2Action, Action dummy3Action, Action<long> overheadAction, Action<long> workloadAction, Job targetJob,
             Action globalSetupAction, Action globalCleanupAction, Action iterationSetupAction, Action iterationCleanupAction, long operationsPerInvoke,
-            bool includeMemoryStats, Encoding encoding, string benchmarkName)
+            bool includeMemoryStats, Encoding encoding, string benchmarkName, CacheClearingStrategy cacheClearingStrategy)
```

Comment:
is something can be created using `targetJob.ResolveValue` you should not add a new parameter to the ctor and  edit the codegen but rather resolve the value as we do for example for `Clock = targetJob.ResolveValue(InfrastructureMode.ClockCharacteristic, Resolver);`